conf/cmake/ConfigureDependencies.cmake

```
@@ -3,167 +3,111 @@
+    set_target_properties(BLAS::BLAS PROPERTIES INTERFACE_LINK_LIBRARIES  "${LAPACK_LIBRARIES}" INTERFACE_LINK_FLAGS "${LAPACK_LINKER_FLAGS}" )
 endif()
 
+
+
 ### OPTIONAL DEPENDENCIES ###
 #############################
 
 if(hpGEM_USE_MPI)
-    FIND_PACKAGE(MPI REQUIRED)
-    # Add OMPI_SKIP_MPICXX to exclude the deprecated c++ bindings of MPI
-    add_definitions(-DHPGEM_USE_MPI -DOMPI_SKIP_MPICXX)
-    include_directories(${MPI_CXX_INCLUDE_PATH})
-    if(NOT MPI_FOUND)
-        message(FATAL_ERROR "The option you have chosen requires mpi and you do not have this installed. Please install")
-    endif()
-    # Create target for easy linking
-    if(NOT TARGET MPI::MPI_CXX)
-        # Note, from CMAKE 3.9 there is a proper target defined
-        add_library(MPI::MPI_CXX INTERFACE IMPORTED)
-        target_include_directories(MPI::MPI_CXX INTERFACE ${MPI_CXX_INCLUDE_PATH})
-        target_link_libraries(MPI::MPI_CXX INTERFACE "${MPI_CXX_LIBRARIES}")
-        target_compile_options(MPI::MPI_CXX INTERFACE "${MPI_CXX_COMPILE_FLAGS}")
-        target_link_options(MPI::MPI_CXX INTERFACE "${MPI_CXX_LINK_FLAGS}")
-    endif()
+    FIND_PACKAGE(MPI REQUIRED COMPONENTS CXX)
+    add_definitions(-DHPGEM_USE_MPI)
 endif()
 
 if(hpGEM_USE_METIS)
     FIND_PACKAGE(METIS REQUIRED)
     add_definitions(-DHPGEM_USE_METIS)
     include_directories(${METIS_INCLUDE_DIR})
-    if(NOT METIS_FOUND)
-        message(FATAL_ERROR "The option you have chosen requires metis and you do not have this installed. Please install")
-    endif()
     # Create target for easy linking
     add_library(METIS::METIS INTERFACE IMPORTED)
-    target_include_directories(METIS::METIS INTERFACE "${METIS_INCLUDE_DIR}")
-    target_link_libraries(METIS::METIS INTERFACE "${METIS_LIBRARIES}")
+    set_target_properties(METIS::METIS PROPERTIES 
+    INTERFACE_LINK_LIBRARIES "${METIS_LIBRARIES}"
+    INTERFACE_INCLUDE_DIRECTORIES "${METIS_INCLUDE_DIR}")
 endif()
 
 if(hpGEM_USE_QHULL)
     FIND_PACKAGE(QHULL REQUIRED)
     add_definitions(-DHPGEM_USE_QHULL)
-    if(NOT QHULL_FOUND)
-        message(FATAL_ERROR "The option you have chosen requires QHull and you do not have this installed. Please install")
-    endif()
     # Create target for easy linking
     add_library(QHULL::QHULL INTERFACE IMPORTED)
-    target_include_directories(QHULL::QHULL INTERFACE "${QHULL_INCLUDE_DIR}")
     if(CMAKE_BUILD_TYPE MATCHES "Debug")
-        target_link_libraries(QHULL::QHULL INTERFACE "${QHULL_DEBUG_LIBRARIES}")
+        set_target_properties(QHULL::QHULL PROPERTIES 
+        INTERFACE_LINK_LIBRARIES "${QHULL_DEBUG_LIBRARIES}" 
+        INTERFACE_INCLUDE_DIRECTORIES "${QHULL_INCLUDE_DIR}")
     else()
-        target_link_libraries(QHULL::QHULL INTERFACE "${QHULL_LIBRARIES}")
+        set_target_properties(QHULL::QHULL PROPERTIES 
+        INTERFACE_LINK_LIBRARIES "${QHULL_LIBRARIES}" 
+        INTERFACE_INCLUDE_DIRECTORIES "${QHULL_INCLUDE_DIR}")
```

```suggestion
        set_target_properties(QHULL::QHULL PROPERTIES 
            INTERFACE_LINK_LIBRARIES "${QHULL_LIBRARIES}" 
            INTERFACE_INCLUDE_DIRECTORIES "${QHULL_INCLUDE_DIR}")
```
Indentation suggestion